lib/rucio/common/stomp_utils.py

```diff
@@ -15,145 +15,415 @@
+                             str(error))
+                continue
+            except Exception as error:
+                self._logger(logging.ERROR, "[broker] Could not deliver message: %s", str(error))
+                continue
+
+            msg_event_type = str(message["event_type"]).lower()
+            msg_payload = message.get("payload", {})
+            if msg_event_type.startswith("transfer") or msg_event_type.startswith("stagein"):
+                self._logger(logging.DEBUG,
+                             "[broker] - event_type: %s, scope: %s, name: %s, rse: %s, request-id: %s, "
+                             "transfer-id: %s, created_at: %s",
+                             msg_event_type,
+                             msg_payload.get("scope", None),
+                             msg_payload.get("name", None),
+                             msg_payload.get("dst-rse", None),
+                             msg_payload.get("request-id", None),
+                             msg_payload.get("transfer-id", None),
+                             str(message["created_at"]))
+
+            elif msg_event_type.startswith("dataset"):
+                self._logger(logging.DEBUG,
+                             "[broker] - event_type: %s, scope: %s, name: %s, rse: %s, rule-id: %s, created_at: %s)",
+                             msg_event_type,
+                             msg_payload.get("scope", None),
+                             msg_payload.get("name", None),
+                             msg_payload.get("rse", None),
+                             msg_payload.get("rule_id", None),
+                             str(message["created_at"]))
+
+            elif msg_event_type.startswith("deletion"):
+                if "url" not in msg_payload:
+                    msg_payload["url"] = "unknown"
+                self._logger(logging.DEBUG,
+                             "[broker] - event_type: %s, scope: %s, name: %s, rse: %s, url: %s, created_at: %s)",
+                             msg_event_type,
+                             msg_payload.get("scope", None),
+                             msg_payload.get("name", None),
+                             msg_payload.get("rse", None),
+                             msg_payload.get("url", None),
+                             str(message["created_at"]))
+            else:
+                self._logger(logging.DEBUG, "[broker] Other message: %s", message)
+
+        return to_delete
+
+    def subscribe(self, id_: str, ack: str, destination: None | str = None, **kwargs):
+        """
+        Subscribe
+
+        Args:
+            id_ (str): The identifier to uniquely identify the subscription
+            ack (str): Either auto, client or client-individual
+            destination (None | str, optional): The topic or queue to subscribe to. If None then
+                                                destination is taken from the rucio config Defaults to None.
```

Comment:
Please avoid specifying the argument types in the docstring if they are already present as type annotations; there's no need to duplicate this information and it can cause mismatches between the type annotation and the type specified in the docstring if changes are made down the line